app/controllers/base_assets_controller.rb

```diff
@@ -24,8 +24,16 @@ def base_asset_params
         asset[:redirect_url] = nil
       end
 
-      if asset.has_key?(:access_limited) && asset[:access_limited].empty?
-        asset[:access_limited] = []
+      if asset.has_key?(:access_limited) && asset[:access_limited].present?
+        asset[:access_limited_user_ids] = asset[:access_limited]
+      end
+
+      if asset.has_key?(:access_limited_user_ids) && asset[:access_limited_user_ids].empty?
```

Comment:
I think we're missing the old 'reset' behaviour when the old `access_limited` param is empty? Looks like there's no longer a (reset) test for it either, but since the attribute is still allowed we should support this case as well. We should be able to do it in one like this using `to_a`.

Reply:
Yup! So I think the "reset" behaviour applies to the new `access_limited_user_ids` with this if statement. And the old  `access_limited` param also adopts this behaviour as its simply mapped to `access_limited_user_ids` in lines 27 to 29. Should be covered by this [test](https://github.com/alphagov/asset-manager/pull/677/files#diff-d438e8bf052be84189bf1bcbed65e9e4R35).

Reply:
Hmm...I'm still unsure and I'll try to explain why:

   - In the code, we only do the mapping behaviour if the old `access_limited` param is `present?`, but the edge case here is when it's `''`, which isn't "present"
   - In the [test](https://github.com/alphagov/asset-manager/pull/677/files#diff-d438e8bf052be84189bf1bcbed65e9e4R35), I think we're just seeing [the default behaviour for the asset](https://github.com/alphagov/asset-manager/blob/master/app/models/asset.rb#L35) (i.e. the existing test was dodgy)

The test that's notably missing is [the reset one for `access_limited_user_ids`](https://github.com/alphagov/asset-manager/pull/677/files#diff-d438e8bf052be84189bf1bcbed65e9e4R233), but using the old name. Am I missing something?

Reply:
Yup - I've fixed resetting an existing parameter using a blank string on the access_limit param and added test case.